extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java

```diff
@@ -97,12 +98,23 @@ public void handle(Throwable t) {
                         // TODO: is there a better way todo retry?
                         // TODO: should we only retry on a specific errors?
 
+                        // try a 2nd time
```

Comment:
It might be simpler to use Mutiny instead. Thus you can configure the number of retry but also the exponential backoff.

I'm not sure if the (Vertx) context matters here but a simple version could be:

Uni.createFrom().future(() -> request.send().toComplerableStage()).onFailure().retry().atMost(x).withBackoff(...)

Reply:
Updated